include/rocksdb/memtablerep.h

```diff
@@ -81,31 +81,49 @@ class MemTableRep {
   // single buffer and pass that in as the parameter to Insert).
   // REQUIRES: nothing that compares equal to key is currently in the
   // collection, and no concurrent modifications to the table in progress
-  //
+  virtual void Insert(KeyHandle handle) = 0;
+
+  // Same as ::Insert
   // Returns false if MemTableRepFactory::CanHandleDuplicatedKey() is true and
   // the <key, seq> already exists.
-  virtual bool Insert(KeyHandle handle) = 0;
+  virtual bool InsertAndReturn(KeyHandle handle) {
```

Comment:
s/InsertAndReturn/InsertUnique/ and same for the rest?

Reply:
I like the *Unique idea but I am concerned that i would cause confusion as *Unique could return true even if the underlying structure does not support checking for uniqueness. And I would like to avoid the overhead of a if-then-else branch in every ::Insert* invocation.

Reply:
I like *AndReturn, it's nice and explicit.

Reply:
`InsertAndHandleDuplication`?

Reply:
We had an offline chat and ended up agreeing on InsertKey()